noise/conn.go

```diff
@@ -351,6 +355,21 @@ func (c *Conn) IsRemoteAuthenticated() bool {
 	return c.isRemoteAuthenticated
 }
 
+// StaticKey returns the static key of the remote peer. It is useful in case the
+// static key is only transmitted during the handshake.
+func (c *Conn) StaticKey() ([]byte, error) {
+	if !c.IsRemoteAuthenticated() {
+		return nil, errors.New("noise: remote peer not authenticated")
+	}
+	if !c.handshakeComplete {
```

Comment:
is it important that the handshake has to be finished?

Reply:
Mhhhh I wanna make sure that the key I get when calling `StaticKey` is the right one (i.e. verified by `PublicKeyVerifier`). Maybe it is possible to deliver it before the handshake completely finishes, but in my use case, I'm fine with waiting the handshake to be finished so I'm sure.
In any case, looking at the code where the `PublicKeyVerifier` gets called, it's almost at the end of the `Handshake` method so not even sure what we can do "earlier".